src/Gaufrette/Adapter/GoogleCloudClientStorage.php

```diff
@@ -0,0 +1,295 @@
+<?php
+namespace Gaufrette\Adapter;
+
+use Gaufrette\Adapter;
+use Gaufrette\Adapter\MetadataSupporter;
+use Gaufrette\Adapter\ListKeysAware;
+use Gaufrette\Exception\FileNotFound;
+use Gaufrette\Exception\StorageFailure;
+use Google\Cloud\Exception\NotFoundException;
+use Google\Cloud\Storage\Bucket;
+use Google\Cloud\Storage\StorageClient;
+use Google\Cloud\Storage\StorageObject;
+
+/**
+ * Google Cloud Storage adapter using the Google Cloud Client Library for PHP
+ * http://googlecloudplatform.github.io/google-cloud-php/
+ *
+ * @package Gaufrette
+ * @author  Lech Buszczynski <lecho@phatcat.eu>
+ */
+final class GoogleCloudClientStorage implements Adapter, MetadataSupporter, ListKeysAware
+{
+    /**
+     * @var StorageClient
+     */
+    private $storageClient;
+
+    /**
+     * @var Bucket
+     */
+    private $bucket;
+    private $options      = array();
+    private $metadata     = array();
+
+    /**
+     * @param StorageClient    $service    Authenticated storage client class
+     * @param string           $bucketName Name of the bucket
+     * @param array            $options    Options are: "directory" and "acl" (see https://cloud.google.com/storage/docs/access-control/lists)
+     */
+    public function __construct(StorageClient $storageClient, $bucketName, $options = array())
+    {
+        $this->storageClient = $storageClient;
+        $this->initBucket($bucketName);
```

Comment:
I'm not sure about this one. I think it's done by other adapters too but it's totally questionable: not only a bucket has to be created but it also has to be properly configured (lifecycle & retention policies, IAM bindings, etc...) and this is way beyond Gaufrette feature set IMHO.

EDIT: Or I misread the code and this only initialize `bucket` property? :thinking:

Reply:
Yes it initializes the `bucket` property with a lazy objects from the SDK (no requests made to the backend), cf https://sourcegraph.com/github.com/googleapis/google-cloud-php/-/blob/Storage/src/StorageClient.php#L125